scipy/special/orthogonal.py

```diff
@@ -525,12 +532,277 @@ def h_roots(n, mu=False):
     if n < 1 or n != m:
         raise ValueError("n must be a positive integer.")
 
-    mu0 = np.sqrt(np.pi)
-    an_func = lambda k: 0.0*k
-    bn_func = lambda k: np.sqrt(k/2.0)
-    f = cephes.eval_hermite
-    df = lambda n, x: 2.0 * n * cephes.eval_hermite(n-1, x)
-    return _gen_roots_and_weights(m, mu0, an_func, bn_func, f, df, True, mu)
+    if n <= 200:
+        mu0 = np.sqrt(np.pi)
+        an_func = lambda k: 0.0*k
+        bn_func = lambda k: np.sqrt(k/2.0)
+        f = cephes.eval_hermite
+        df = lambda n, x: 2.0 * n * cephes.eval_hermite(n-1, x)
+        return _gen_roots_and_weights(m, mu0, an_func, bn_func, f, df, True, mu)
+    else:
+        nodes, weights = h_roots_asy(n)
+        if mu:
+            return nodes, weights, sum(weights)
+        else:
+            return nodes, weights
+
+
+def compute_tauk(n, k, steps=5):
+    """Helper function for lemma 3.1
```

Comment:
> I guess it's for Townsend, A. and Trogdon, T. and Olver, S. (2014) whose reference is defined in the docstring for a different function.

Of course it is for this paper that defines the whole algorithm.
Where should this top-level reference go? Up to the other two
in this module?

I'll write better docstrings tomorrow. I agree this is to little to be useful.